pkg/controllers/dynakube/extension/eec/reconciler.go

```diff
@@ -70,3 +83,87 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 
 	return r.createOrUpdateStatefulset(ctx)
 }
+
+func (r *reconciler) reconcileTlsSecret(ctx context.Context) error {
+	query := k8ssecret.Query(r.client, r.client, log)
+
+	secret, err := query.Get(ctx, types.NamespacedName{
+		Name:      consts.GetTlsSecretName(r.dk.Name),
+		Namespace: r.dk.Namespace,
+	})
+
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+
+	if r.dk.Spec.Templates.ExtensionExecutionController.TlsRefName != "" {
```

Comment:
Saw that this is used several times, should we abstract it for the sake of readability?

Reply:
good catch, I saw we do it via method on dk.

Reply:
https://github.com/Dynatrace/dynatrace-operator/pull/3778/commits/c1fb51c95e19d48604f6c04fc85b977cb720dace